Allow "browsing" styles through the browser panel

nyalldawson opened the pull request:
Double clicking a style .xml in the browser now opens the manager dialog, allowing browsing and non-edit actions for the style. Additionally there's a new "copy items to default style" action available, which allows users to easily transfer symbols and ramps from these other .xml files to their own local style database.

![peek 2019-01-15 11-16](https://user-images.githubusercontent.com/1829991/51151548-17ddbb80-18b7-11e9-954c-fd97e196c1e8.gif)

nyalldawson committed:
Add a read only mode to QgsStyleManagerDialog

Allows the dialog to browse an existing style, but prevents users
from making any edits to the style.

nyalldawson committed:
Add method to hide favorites and smart groups from style manager dialog

nyalldawson committed:
Fix initial selected tab in style manager dialog

nyalldawson committed:
[FEATURE] Double clicking a style .xml in the browser now opens
the manager dialog, allowing browsing and non-edit actions for the
style.

nyalldawson committed:
Refactor

nyalldawson committed:
[FEATURE] When browsing a style from the browser panel, allow copying
symbols to the default style from the right click menu and share button

Allows users to easily transfer symbols and ramps from a style xml
file directly to their local style database.

nyalldawson committed:
Release busy cursor before showing message box

nyalldawson commented:
I don't show it in the cast, but it's also shown in the Import/Export button menu. Does that satisfy the discoverability issue?

NathanW2 commented:
Nifty!!  I wonder if we can tell for the browser if the file is a QGIS style and not some other random xml file so we can show a better icon or even  the symbols inline in the browser ;) ;)

nyalldawson commented:
@nathanw2 we already do sniff the XML to checking it's a style. The duplicate entry comes from the gdal provider, which treats all XML files as a layer source...

nyalldawson commented:
And that makes me think I should just solve that instead :p

nyalldawson committed:
Rename action for clarity

nyalldawson committed:
Considerably simplify code, remove duplication

nyalldawson committed:
Nicer strings

nyalldawson committed:
Allow additional tags to be set when copying items to default style

nyalldawson committed:
Add 'Copy to default' action as a dedicated button when browsing non-default style

nyalldawson committed:
HIG

nyalldawson commented:
@nirvn I've addressed your concerns now. I've left the dedicated import/export button, because that has some other useful stuff like the ability to export symbols as images, but I've also added the dedicated explicit button for copying items to the default style.

nirvn commented:
Thanks for the flexibility. +1 to merge
